presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1894,8 +1895,25 @@ public ConnectorPushdownFilterResult pushdownFilter(ConnectorSession session, Co
             }
         }
 
+        SchemaTableName tableName = ((HiveTableHandle) tableHandle).getSchemaTableName();
         HivePartitionResult hivePartitionResult = partitionManager.getPartitions(metastore, tableHandle, constraint, session);
 
+        //if TupleDomain isNone or optmized remaningExpression is false return ConnectorPushdownFilterResult with TupleDomain.none() layout
+        if (decomposedFilter.getTupleDomain().isNone()) {
+            return new ConnectorPushdownFilterResult(
```

Comment:
"I would also update PickTableLayout to remove the logic of extracting TupleDomain and checking for none."  This will result in assertion failure in MetadaManager.pushdownFilter